superset-frontend/src/explore/components/controls/DndColumnSelectControl/utils/optionSelector.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { ColumnMeta } from '@superset-ui/chart-controls';
+
+export class OptionSelector {
+  groupByOptions: ColumnMeta[];
+
+  options: { string: ColumnMeta };
+
+  isScalar: boolean;
+
+  constructor(
+    options: { string: ColumnMeta },
+    values: string[] | string | null,
+  ) {
+    this.options = options;
+    let groupByValues: string[];
+    if (Array.isArray(values)) {
+      groupByValues = values;
+      this.isScalar = false;
+    } else {
+      groupByValues = values ? [values] : [];
+      this.isScalar = true;
+    }
```

Comment:
This might just be me, but I feel the term `scalar` can cause confusion here (in mathematics a scalar is a real number, which in the context of JS would be `number`). I'd rather use `isArray` to highlight that this refers to distinguishing between a arrays and non-arrays.

Reply:
+1 on change this to `isArray` check since we may want to other more complex `option` in the future (e.g. `AdhocFilter` and `AdhocMetric`).

Reply:
`scalar` in linear algebra, relative to `vector`. refer to SQLAlachemy and Pandas variable naming. 

`isArray` more straightforward, I changed this variable name.